source/test/TestApp/PlayFabEventTest.cpp

```diff
@@ -218,6 +220,56 @@ namespace PlayFabUnit
         std::bind(&PlayFabEventTest::NonStaticEmitEventCallback, this, std::placeholders::_1, std::placeholders::_2));
     }
 
+    void PlayFabEventTest::GenericMultiThreadedTest(uint32_t pNumThreads, uint32_t pNumEventsPerThread)
+    {
+        uint32_t localNumEventsReturned = 0;
+        uint32_t localTotalEventsExpected = pNumThreads * pNumEventsPerThread;
+        for (uint32_t thread = 0; thread < pNumThreads; ++thread)
+        {
+            testThreadPool.emplace_back([&localNumEventsReturned, localTotalEventsExpected, pNumEventsPerThread, this]() {
+                std::shared_ptr<PlayFabEventAPI*> api = SetupEventTest();
+                for (uint32_t i = 0; i < pNumEventsPerThread; ++i)
+                {
+                    (*api)->EmitEvent(MakeEvent(0, PlayFabEventType::Default),
+                    [&localNumEventsReturned, pNumEventsPerThread, localTotalEventsExpected]
+                    (std::shared_ptr<const PlayFab::IPlayFabEvent>, std::shared_ptr<const PlayFab::IPlayFabEmitEventResponse>)
+                    {
+                        localNumEventsReturned++;
+                        if (localNumEventsReturned == localTotalEventsExpected)
+                        {
+                            (*eventTestContext)->Pass("Threaded callback worked as expected!");
+                        }
+                    });
```

Comment:
nit: it's pretty hard to read this EmitEvent call with the lambda placed inline here. Recommend saving it is a named temporary somewhere (preferably outside of all of the loops since you don't need a unique lambda for each event)

```
auto EmitEventCallback = ...;
(*api)->EmitEvent(MakeEvent(0, Default, emitEventCallback));
```